lib/google_crawler/search/keyword_worker.ex

```diff
@@ -0,0 +1,28 @@
+defmodule GoogleCrawler.Search.KeywordWorker do
+  use Oban.Worker,
+    priority: 1,
+    max_attempts: 10,
+    unique: [period: 30]
+
+  alias GoogleCrawler.Repo
+  alias GoogleCrawler.Search.{Keyword, Parser}
+
+  @impl Worker
+  def perform(%Oban.Job{args: %{"keyword_id" => keyword_id}}) do
+    keyword = Repo.get_by!(Keyword, %{id: keyword_id})
+
+    updated_keyword =
+      keyword
+      |> Ecto.Changeset.change(status: :in_progress)
+      |> Repo.update!()
+      |> Ecto.Changeset.change(result: parsed_result(keyword), status: :finished)
+      |> Repo.update!()
+
+    {:ok, updated_keyword}
+  end
+
+  defp parsed_result(keyword) do
```

Comment:
Maybe

```suggestion
  defp parsed_result!(keyword) do
```

as it seems to raise error on unexpected cases.